### ISSUE TITLE
Home Assistant

### ISSUE THREAD

**Issue by ralpho1337** [platysma/vienna-smartmeter#1]
Hi platysma,

Great take on the smartmeter API - thanks for the implementation. Are you planning on building a plugin for home assistant?

Thanks
Ralph

**Comment by platysma** [comment]
Hi ralpho!

Thanks! Yes, that was the initial plan. I started a private repo for the plugin, but since I am new to home assistant and didn't get much done. Do you want to contribute?

**Comment by mariomurrent-softwaresolutions** [comment]
Maybe I would also like to contribute

**Comment by platysma** [comment]
[Here](https://github.com/platysma/hass-vienna-smartmeter) is the public repo for the smartmeter hassio integration. I used [this](https://github.com/oncleben31/cookiecutter-homeassistant-custom-component/issues/124) cookiecutter project to start off and I copied over some files and transformed the existing code into async functions for simple testing and debugging, but I couldn't get it to work so far.

Maybe we should add an async module to this repo and add it as an external requirement to the the integration. I'm not sure if async is the way to go here, but let's discuss this in the other repo.
